job_creator.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+class JobCreator
+{
+    public function getInstallerVersion($githubRepository, $branch)
+    {
+        $repo = explode('/', $githubRepository)[1];
+        if (in_array($repo, NO_INSTALLER_REPOS)) {
+            return '';
+        }
+        $v = explode('.', $branch);
+        if (count($v) == 1) {
+            return '4.x-dev';
+        }
+        if (in_array($repo, LOCKSTEPED_REPOS)) {
+            return '4' . $v[1] . 'x-dev';
+        } else {
+            // use the latest minor version of installer
+            $a = array_keys(INSTALLER_TO_PHP_VERSIONS);
+            $a = array_map(fn($k) => (int) $k, $a);
+            sort($a);
+            return $a[count($a) - 1] . 'x-dev';
+        }
+    }
+    
+    public function createJob($phpNum, $opts)
+    {
+        $v = str_replace('.x-dev', '', $this->installerVersion);
+        $v = $v ?: '4';
+        $phpVersions = INSTALLER_TO_PHP_VERSIONS[$v];
+        $default = [
+            # ensure there's a default value for all possible return keys
+            # this allows use to use `if [ "${{ matrix.key }}" == "true" ]; then` in github-actions-ci-cd/ci.yml
+            'installer_version' => $this->installerVersion,
+            'php' => $phpVersions[$phpNum] ?? $phpVersions[count($phpVersions) - 1],
+            'db' => DB_MYSQL_57,
+            'composer_require_extra' => '',
+            'composer_args' => '',
+            'name_suffix' => '',
+            'phpunit' => false,
+            'phpunit_suite' => 'all',
+            'phplinting' => false,
+            'phpcoverage' => false,
+            'endtoend' => false,
+            'endtoend_suite' => 'root',
+            'endtoend_config' => '',
+            'js' => false,
+        ];
+        return array_merge($default, $opts);
+    }
+    
+    private function parseBoolValue($value)
+    {
+        return ($value === true || $value === 'true');
+    }
+
+    private $installerVersion = null;
+    
+    public function createJson($inputs)
+    {
+        // $myRef will either be a branch for push (i.e cron) and pull-request (target branch), or a semver tag
+        $myRef = $inputs['github_my_ref'];
+        $isTag = preg_match('#^[0-9]+\.[0-9]+\.[0-9]+$#', $myRef, $m);
```

Comment:
```suggestion
        $isTag = preg_match('#^[0-9]+\.[0-9]+\.[0-9]+$#', $myRef, $matches);
```
You'll need to change this wherever it's used below as well.

Reply:
I like small `$m`

Reply:
We discussed this offline. The small `$m` is inoffensive here because it's used immediately after its assignment, which provides clear context to its meaning. If we change the code to use the variable again further down, it should be changed, but as it stands it's okay.